packages/theme-product/theme-product.test.js

```diff
@@ -0,0 +1,188 @@
+/**
+ * @jest-environment jsdom
+ */
+import {
+  getVariant,
+  createOptionArrayFromOptionCollection
+} from "./theme-product";
+import { getProductJsonMock } from "./__mocks__/product";
+
+describe("getVariant()", () => {
+  let productJson;
+
+  beforeEach(() => {
+    let markup = getProductJsonMock();
+    document.body.innerHTML = `<script id="productJson" type="application/json">${markup}</script>`;
+    productJson = JSON.parse(document.getElementById("productJson").innerHTML);
+  });
+
+  test("is a function exported by theme-product.js", () => {
+    expect(typeof getVariant).toBe("function");
+  });
+
+  test("throws an error if parameters are missing", () => {
+    expect(() => {
+      getVariant();
+    }).toThrow();
+
+    expect(() => {
+      getVariant({});
+    }).toThrow();
+
+    expect(() => {
+      getVariant(null, {});
+    }).toThrow();
+  });
+
+  test("throws an error if product json object is empty", () => {
+    expect(() => {
+      getVariant({}, 6908023078973);
+    }).toThrow();
+  });
+
+  test("returns a product variant object when called with valid arguments", () => {
```

Comment:
"returns a product variant object ~when called with valid arguments~ when a match is found"